app/models/patient.rb

```diff
@@ -133,6 +133,152 @@ def latest_transfer
       .where('latest_assessment_at < ? OR latest_assessment_at IS NULL', Time.now.in_time_zone('Eastern Time (US & Canada)').beginning_of_day)
   }
 
+  # Patients who are eligible for reminders
+  #
+  # GENERAL SCOPE OVERVIEW:
+  # - Everything before the OR is essentially checking if the HoH or the patient
+  #   who isn't in a household is eligible on their own.
+  # - Everything within the OR is checking if the HoH has any DEPENDENTS (excluding self)
+  #   that would make the HoH eligible to receive notifications for them.
+  scope :better_reminder_eligible, lambda {
+    joins(:dependents)
```

Comment:
Do you need to have a join with the dependents table for these checks? It should only be needed for the `has_eligible_dependents` (which already does this join), right?

Reply:
If you remove that joins then Rails will complain that the two scopes are incompatible. Even though we do not use the dependents join directly in `better_reminder_eligible` we still have to add the joins here so that it is structurally compatible with the `has_eligible_dependents` part of the scope

```
Error:
PatientNotificationEligibilityTest#test_exposure_send_report_when_no_assessments_and_patient_was_created_more_than_1_day_ago:
ArgumentError: Relation passed to #or must be structurally compatible. Incompatible values: [:joins]
    app/models/patient.rb:152:in `block in <class:Patient>'
    test/models/patient_notification_eligibility_test.rb:29:in `assert_eligible'
    test/models/patient_notification_eligibility_test.rb:411:in `block in <class:PatientNotificationEligibilityTest>'
```